modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ObjectBasedParameterValueResolver.java

```
@@ -67,4 +77,35 @@ public Object getParameterValue(String parameterName) throws ValueResolvingExcep
     }
     throw new ValueResolvingException("Unable to resolve value for the parameter: " + parameterName);
   }
+
+  @Override
+  public Map<String, ValueResolver<? extends Object>> getParameters() {
+    HashMap<String, ValueResolver<? extends Object>> parameters = new HashMap<>();
+    try {
+      addFields(getFields(object.getClass()), parameters);
+
+      for (ParameterGroupModel parameterGroupModel : parameterizedModel.getParameterGroupModels()) {
+        Optional<ParameterGroupModelProperty> modelProperty =
+            parameterGroupModel.getModelProperty(ParameterGroupModelProperty.class);
+        if (modelProperty.isPresent()) {
+          ParameterGroupModelProperty property = modelProperty.get();
+          Field container = (Field) property.getDescriptor().getContainer();
+          Object parameterGroup = getFieldValue(object, container.getName(), reflectionCache);
+          addFields(getFields(parameterGroup.getClass()), parameters);
+        }
+      }
+    } catch (Exception e) {
+      LOGGER.debug("An error occurred trying to obtain the parameters.");
```

don't swallow the exception